gstreamer/src/value_serde.rs

```diff
@@ -0,0 +1,506 @@
+// Copyright (C) 2018 François Laignel <fengalin@free.fr>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use glib;
+use glib::{StaticType, ToValue};
+
+use num_rational::Rational32;
+
+use serde::de;
+use serde::de::{Deserialize, Deserializer, SeqAccess, Visitor};
+use serde::ser;
+use serde::ser::{Serialize, SerializeTuple, Serializer};
+
+use std::{fmt, mem};
+
+use DateTime;
+use Sample;
+
+use value::*;
+
+pub const ARRAY_TYPE_NAME: &'static str = "Array";
+pub const LIST_TYPE_NAME: &'static str = "List";
+
+fn get_other_type_id<T: StaticType>() -> usize {
+    match T::static_type() {
+        glib::Type::Other(type_id) => type_id,
+        type_ => panic!("Expecting `Other` variant, found `{}`", type_),
+    }
+}
+
+lazy_static! {
+    pub(crate) static ref ARRAY_OTHER_TYPE_ID: usize = get_other_type_id::<Array>();
+    pub(crate) static ref BITMASK_OTHER_TYPE_ID: usize = get_other_type_id::<Bitmask>();
+    pub(crate) static ref DATE_TIME_OTHER_TYPE_ID: usize = get_other_type_id::<DateTime>();
+    pub(crate) static ref FRACTION_OTHER_TYPE_ID: usize = get_other_type_id::<Fraction>();
+    pub(crate) static ref FRACTION_RANGE_OTHER_TYPE_ID: usize =
+        get_other_type_id::<FractionRange>();
+    pub(crate) static ref INT_RANGE_I32_OTHER_TYPE_ID: usize = get_other_type_id::<IntRange<i32>>();
+    pub(crate) static ref INT_RANGE_I64_OTHER_TYPE_ID: usize = get_other_type_id::<IntRange<i64>>();
+    pub(crate) static ref LIST_OTHER_TYPE_ID: usize = get_other_type_id::<List>();
+    pub(crate) static ref SAMPLE_OTHER_TYPE_ID: usize = get_other_type_id::<Sample>();
+}
+
+impl<'a> Serialize for Fraction {
```

Comment:
This one can't be derived?

Reply:
It could, but that would lead to things like `((1, 3))` due to the `Rational32` being an inner type. This also appears in some `FormattedValue`s, we might want to avoid these too (but that would lessen the use cases for my modification in `gir` :)).

Reply:
I see :)